reactive/kotlinx-coroutines-rx2/src/RxAwait.kt

```diff
@@ -43,12 +51,23 @@ public suspend fun CompletableSource.await(): Unit = suspendCancellableCoroutine
  */
 @Suppress("UNCHECKED_CAST")
 public suspend fun <T> MaybeSource<T>.awaitSingleOrNull(): T? = suspendCancellableCoroutine { cont ->
-    subscribe(object : MaybeObserver<T> {
-        override fun onSubscribe(d: Disposable) { cont.disposeOnCancellation(d) }
-        override fun onComplete() { cont.resume(null) }
-        override fun onSuccess(t: T) { cont.resume(t) }
-        override fun onError(error: Throwable) { cont.resumeWithException(error) }
-    })
+    subscribe(object : MaybeObserver<T & Any> {
```

Comment:
```suggestion
    subscribe(object : MaybeObserver<T> {
```
Then, the cast can be removed, along with the suppression.